virttest/utils_test/__init__.py

```diff
@@ -838,205 +848,284 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
                 logging.error("Avocado git installation failed")
                 return False
-            for plugin in plugins[installtype]:
+            for plugin in self.plugins[self.installtype]:
                 cmd = "[ -d optional_plugins/%s ] && cd optional_plugins/" % plugin
-                cmd += "%s && %s setup.py install && cd .." % (plugin, python)
-                if session.cmd_status(cmd) > 0:
+                cmd += "%s && %s setup.py install && cd .." % (plugin, self.python)
+                if self.session.cmd_status(cmd) != 0:
                     logging.error("Avocado plugin %s git "
                                   "installation failed", plugin)
                     return False
+            if self.avocado_vt and not self.git_install(self.avocado_vt_repo,
+                                                        make='requirements'):
+                logging.error("Avocado-VT git installation failed")
+                return False
         return True
 
-    def runtest(session, testrepo, testlist, timeout, result_path, test_path):
+    def git_install(self, repo_path, install=True, make=None):
         """
-        run test
+        method to clone and install from git repo inside guest
+
+        :param repo_path: repo link to be cloned
         """
-        success = True
 
-        cmd = "[ -n \"$(ls %s)\" ]" % test_path
-        if session.cmd_status(cmd) > 0:
-            cmd = "mkdir -p %s" % test_path
-            session.cmd(cmd)
+        cmd = "cd %s;" % self.test_path
+        cmd += "rm -rf %s" % self.repo_name(repo_path)
+        self.session.cmd(cmd)
+        cmd = "git clone %s;" % repo_path
+        cmd += "cd %s;" % self.repo_name(repo_path)
+        if make:
+            cmd += "make %s;" % make
+        if install:
+            cmd += "%s setup.py install" % self.python
+        return self.session.cmd_status(cmd) == 0
 
-        cmd = "[ -n \"$(ls %s)\" ]" % result_path
-        if session.cmd_status(cmd) > 0:
-            cmd = "mkdir -p %s" % result_path
-            session.cmd(cmd)
+    def repo_name(self, repo_path):
+        """ Wrapper to return the repo name """
+        return repo_path.split(os.sep)[-1].split(".")[0]
 
+    def runtest(self):
+        """
+        Run test method to download the tests and trigger avocado command
+        """
         logging.debug("Downloading Test")
-        cmd = "cd %s;[ -d %s ] || git clone %s" % (test_path,
-                                                   testrepo.split("/")[-1].split('.')[0],
-                                                   testrepo)
-        status, output = session.cmd_status_output(cmd, timeout=100)
-        if status > 0:
-            raise exceptions.TestError("Downloading test failed: %s" % output)
+        if self.avocado_vt:
+            cmd = "avocado vt-bootstrap --vt-type %s" % self.vt_type
+            if self.guest_image:
+                cmd += " --vt-guest-os %s --yes-to-all" % self.guest_image
```

Comment:
move `--yes-to-all`  to above cmd= statement so that even without self.guest_image we go by default downloading ...